manifests/install/remove_packages_osx.pp

```diff
@@ -0,0 +1,60 @@
+# == Class puppet_agent::install::remove_packages_osx
+#
+# Sadly, special handling is required to clear up puppet_enterprise installation
+# on 3.8.
+#
+class puppet_agent::install::remove_packages_osx {
+  assert_private()
+
+  if $::puppet_agent::is_pe {
+    # shutdown services
+    service { 'pe-puppet':
+      ensure => stopped,
+    }->
+    service { 'pe-mcollective':
+      ensure => stopped,
+    }->
+
+    # remove old users and groups
+    user { 'pe-puppet':
+      ensure => absent,
+    }->
+    user { 'pe-mcollective':
+      ensure => absent,
+    }->
+
+    # remove old /opt/puppet files
+    file { '/opt/puppet':
+      ensure => absent,
+      force  => true,
+      backup => false,
+    }
```

Comment:
As an alternative to the above, we could do something delightful like an Exec of:

``` sh
for old_puppet_file in $(pkgutil --pkgs | grep puppet | xargs -Isub pkgutil --files sub | sort -r | uniq | grep '^opt/puppet'); do
  
  if [ -d "/${old_puppet_file?}" ]; then
    rmdir "/${old_puppet_file?}"
  elif [ -f "/${old_puppet_file?}" ]; then
    rm "/${old_puppet_file?}"
  fi
done
```

To preserve files added by the user; this would leave behind /opt/puppet/libexec/mcollective, and /opt/puppet/ssl/man/man3 (which presumably gets built post install.  The mcollective plugins are about the only reason I can think of to prefer this route, as this preserves custom plugins that might have been added by the user.

@highb @haus any opinions?